src/graph/ChcGraph.cc

```diff
@@ -599,4 +601,126 @@ std::unique_ptr<ChcDirectedHyperGraph> ChcDirectedHyperGraph::makeEmpty(Logic &
         std::move(predicateRepresentation),
         logic
         );
+}
+
+WitnessInfo ChcDirectedGraph::contractConnectedVertexes(std::vector<EId> edges) {
```

Comment:
```suggestion
WitnessInfo ChcDirectedGraph::contractConnectedVertices(std::vector<EId> edges) {
```

Also, any reason why this cannot take `const &` or `&&` instead of a copy of the loop?

Reply:
Fixed